src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -204,10 +221,20 @@ public void buttonSaveClicked() {
     }
     fc.getExtensionFilters().add(new ExtensionFilter("JSON files", "*.json"));
     fc.setTitle("Save Config");
+    return fc.showSaveDialog(null);
+  }
 
-    File saveFile = fc.showSaveDialog(null);
+  private void saveFile(File saveFile) {
+    JsonParser.createFileFromRawContestConfig(saveFile, createRawContestConfig());
+    // Necessary to keep GUI aligned when default values subbed in if invalid during save process
```

Comment:
Not quite sure what this means.

Reply:
Clarifying comment. Basically if you save a file with an invalid value (like null for number of decimal places), the save procedure replaces it with the default value. Unless we reload the file, the GUI will still have a null value in that field. The reload makes sure the default value populates the text field in this situation.